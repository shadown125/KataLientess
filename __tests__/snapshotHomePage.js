import React from 'react';
import renderer from 'react-test-renderer';
import * as nextRouter from "next/router";
import {SessionProvider} from "next-auth/react";
import HomePage from '../pages/index';

it('Renders home page unchanged', () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: '/' }));

    const tree = renderer.create(
        <SessionProvider>
            <HomePage />
        </SessionProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
})