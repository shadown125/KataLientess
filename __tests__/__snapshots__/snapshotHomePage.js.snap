// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders home page unchanged 1`] = `
Array [
  <div
    className="wrapper"
  >
    <div
      className="app-container"
    >
      <header
        className="page-header"
      >
        <div
          className="container"
        >
          <a
            className="image-wrapper"
            href="https://www.akalientess.com/"
            rel="external noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Logo"
              src="/brandLogo.png"
            />
          </a>
          <h1
            className="headline h1"
          >
            KataLientesS
          </h1>
          <ul
            className="actions"
          >
            <li>
              <a
                className="button button--big icon-configs"
                href="/settings"
                onClick={[Function]}
                onMouseEnter={[Function]}
                type="button"
              />
            </li>
          </ul>
          <button
            className="button button--medium icon-menu"
            onClick={[Function]}
            type="button"
          >
            <span>
              Menu
            </span>
          </button>
        </div>
      </header>
      <main>
        <div
          className="page-body"
        >
          <aside
            className="nav-aside"
          >
            <nav
              className="main-navigation"
            >
              <div
                className="profile is-loading"
              >
                <div
                  className="image-wrapper"
                >
                  <div
                    className="image"
                  />
                </div>
                <div
                  className="content"
                >
                  <div />
                  <div />
                </div>
              </div>
              <ul
                className="actions"
              >
                <li>
                  <a
                    className="link icon-link icon-note"
                    href="/"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <span>
                      Today
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    className="link icon-link icon-note-checked"
                    href="/done-todos"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <span>
                      Done todos
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    className="link icon-link icon-configs"
                    href="/settings"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <span>
                      Settings
                    </span>
                  </a>
                </li>
                <li>
                  <button
                    className="link icon-link icon-logout"
                    onClick={[Function]}
                  >
                    <span>
                      Logout
                    </span>
                  </button>
                </li>
              </ul>
              <div
                className="credits"
              >
                Â© 
                2022
                 All rights reserved by Dawid Oleksiuk
              </div>
            </nav>
          </aside>
          <section
            className="todos-container"
          >
            <div
              className="progression-bar is-loading"
            >
              <div
                className="date"
              >
                <span
                  className="day"
                >
                  23
                </span>
                <span
                  className="month"
                >
                  sty
                </span>
              </div>
              <div
                className="content"
              >
                <h2
                  className="headline h4"
                >
                  Today Tasks
                </h2>
                <div
                  className="bar"
                >
                  <div
                    className="progressed-bar"
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                </div>
                <div
                  className="descriptions"
                >
                  <span
                    className="done-todos"
                  >
                    <span>
                      Done Todos: 0
                    </span>
                  </span>
                  <span
                    className="in-progress-todos"
                  >
                    <span>
                      In progress Todos: 0
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              className="todos is-loading"
            >
              <div
                className="head-container"
              >
                <h2
                  className="headline h5"
                >
                  In Progression
                </h2>
                <ul>
                  <li
                    className="add-todo"
                  >
                    <button
                      className="button button-primary"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Add Todo
                      </span>
                    </button>
                  </li>
                </ul>
              </div>
              <div
                className="main-container"
              >
                <ul
                  className="todo-list"
                >
                  <li>
                    <div
                      className="header"
                    >
                      <div
                        className="headline-loading"
                      >
                        <h3
                          className="headline h5"
                        >
                          Lorem ipsum
                        </h3>
                      </div>
                      <ul
                        className="actions"
                      >
                        <li>
                          <button
                            className="button"
                            type="button"
                          />
                        </li>
                        <li>
                          <button
                            className="button"
                            type="button"
                          />
                        </li>
                      </ul>
                    </div>
                    <div
                      className="main"
                    >
                      <p>
                        <span>
                          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque culpa
                        </span>
                      </p>
                      <p>
                        <span>
                          deleniti dicta ea eius esse ipsa, iusto minus nemo nobis optio quaerat quam, similique sit veniam voluptatibus?
                        </span>
                      </p>
                    </div>
                  </li>
                  <li>
                    <div
                      className="header"
                    >
                      <div
                        className="headline-loading"
                      >
                        <h3
                          className="headline h5"
                        >
                          Lorem ipsum
                        </h3>
                      </div>
                      <ul
                        className="actions"
                      >
                        <li>
                          <button
                            className="button"
                            type="button"
                          />
                        </li>
                        <li>
                          <button
                            className="button"
                            type="button"
                          />
                        </li>
                      </ul>
                    </div>
                    <div
                      className="main"
                    >
                      <p>
                        <span>
                          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque culpa
                        </span>
                      </p>
                      <p>
                        <span>
                          deleniti dicta ea eius esse ipsa, iusto minus nemo nobis optio quaerat quam, similique sit veniam voluptatibus?
                        </span>
                      </p>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </section>
        </div>
      </main>
    </div>
  </div>,
  <section
    className="new-todo"
  >
    <div
      className="notification"
    >
      <span>
        Your new Todo was added successfully
      </span>
    </div>
    <div
      className="container"
    >
      <button
        className="button button--medium icon-cross"
        onClick={[Function]}
      >
        <span>
          Close button
        </span>
      </button>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <label
          htmlFor="title"
        >
          Title:*
        </label>
        <input
          id="title"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Title for your task"
          type="input"
          value=""
        />
        <label
          htmlFor="description"
        >
          Description:
        </label>
        <textarea
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Description text for your Task (optional)"
          value=""
        />
        <button
          className="button button-primary"
          disabled={false}
          type="submit"
        >
          <span>
            Add Todo
          </span>
        </button>
      </form>
    </div>
  </section>,
  <div
    className="backdrop"
  />,
]
`;

exports[`renders home page unchanged 1`] = `
Array [
  <div
    className="wrapper"
  >
    <div
      className="app-container"
    >
      <header
        className="page-header"
      >
        <div
          className="container"
        >
          <a
            className="image-wrapper"
            href="https://www.akalientess.com/"
            rel="external noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Logo"
              src="/brandLogo.png"
            />
          </a>
          <h1
            className="headline h1"
          >
            KataLientesS
          </h1>
          <ul
            className="actions"
          >
            <li>
              <a
                className="button button--big icon-configs"
                href="/settings"
                onClick={[Function]}
                onMouseEnter={[Function]}
                type="button"
              />
            </li>
          </ul>
          <button
            className="button button--medium icon-menu"
            onClick={[Function]}
            type="button"
          >
            <span>
              Menu
            </span>
          </button>
        </div>
      </header>
      <main>
        <div
          className="page-body"
        >
          <aside
            className="nav-aside"
          >
            <nav
              className="main-navigation"
            >
              <div
                className="profile is-loading"
              >
                <div
                  className="image-wrapper"
                >
                  <div
                    className="image"
                  />
                </div>
                <div
                  className="content"
                >
                  <div />
                  <div />
                </div>
              </div>
              <ul
                className="actions"
              >
                <li>
                  <a
                    className="link icon-link icon-note"
                    href="/"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <span>
                      Today
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    className="link icon-link icon-note-checked"
                    href="/done-todos"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <span>
                      Done todos
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    className="link icon-link icon-configs"
                    href="/settings"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <span>
                      Settings
                    </span>
                  </a>
                </li>
                <li>
                  <button
                    className="link icon-link icon-logout"
                    onClick={[Function]}
                  >
                    <span>
                      Logout
                    </span>
                  </button>
                </li>
              </ul>
              <div
                className="credits"
              >
                Â© 
                2022
                 All rights reserved by Dawid Oleksiuk
              </div>
            </nav>
          </aside>
          <section
            className="todos-container"
          >
            <div
              className="progression-bar is-loading"
            >
              <div
                className="date"
              >
                <span
                  className="day"
                >
                  23
                </span>
                <span
                  className="month"
                >
                  sty
                </span>
              </div>
              <div
                className="content"
              >
                <h2
                  className="headline h4"
                >
                  Today Tasks
                </h2>
                <div
                  className="bar"
                >
                  <div
                    className="progressed-bar"
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                </div>
                <div
                  className="descriptions"
                >
                  <span
                    className="done-todos"
                  >
                    <span>
                      Done Todos: 0
                    </span>
                  </span>
                  <span
                    className="in-progress-todos"
                  >
                    <span>
                      In progress Todos: 0
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              className="todos is-loading"
            >
              <div
                className="head-container"
              >
                <h2
                  className="headline h5"
                >
                  In Progression
                </h2>
                <ul>
                  <li
                    className="add-todo"
                  >
                    <button
                      className="button button-primary"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Add Todo
                      </span>
                    </button>
                  </li>
                </ul>
              </div>
              <div
                className="main-container"
              >
                <ul
                  className="todo-list"
                >
                  <li>
                    <div
                      className="header"
                    >
                      <div
                        className="headline-loading"
                      >
                        <h3
                          className="headline h5"
                        >
                          Lorem ipsum
                        </h3>
                      </div>
                      <ul
                        className="actions"
                      >
                        <li>
                          <button
                            className="button"
                            type="button"
                          />
                        </li>
                        <li>
                          <button
                            className="button"
                            type="button"
                          />
                        </li>
                      </ul>
                    </div>
                    <div
                      className="main"
                    >
                      <p>
                        <span>
                          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque culpa
                        </span>
                      </p>
                      <p>
                        <span>
                          deleniti dicta ea eius esse ipsa, iusto minus nemo nobis optio quaerat quam, similique sit veniam voluptatibus?
                        </span>
                      </p>
                    </div>
                  </li>
                  <li>
                    <div
                      className="header"
                    >
                      <div
                        className="headline-loading"
                      >
                        <h3
                          className="headline h5"
                        >
                          Lorem ipsum
                        </h3>
                      </div>
                      <ul
                        className="actions"
                      >
                        <li>
                          <button
                            className="button"
                            type="button"
                          />
                        </li>
                        <li>
                          <button
                            className="button"
                            type="button"
                          />
                        </li>
                      </ul>
                    </div>
                    <div
                      className="main"
                    >
                      <p>
                        <span>
                          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab accusamus alias atque culpa
                        </span>
                      </p>
                      <p>
                        <span>
                          deleniti dicta ea eius esse ipsa, iusto minus nemo nobis optio quaerat quam, similique sit veniam voluptatibus?
                        </span>
                      </p>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </section>
        </div>
      </main>
    </div>
  </div>,
  <section
    className="new-todo"
  >
    <div
      className="notification"
    >
      <span>
        Your new Todo was added successfully
      </span>
    </div>
    <div
      className="container"
    >
      <button
        className="button button--medium icon-cross"
        onClick={[Function]}
      >
        <span>
          Close button
        </span>
      </button>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <label
          htmlFor="title"
        >
          Title:*
        </label>
        <input
          id="title"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Title for your task"
          type="input"
          value=""
        />
        <label
          htmlFor="description"
        >
          Description:
        </label>
        <textarea
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Description text for your Task (optional)"
          value=""
        />
        <button
          className="button button-primary"
          disabled={false}
          type="submit"
        >
          <span>
            Add Todo
          </span>
        </button>
      </form>
    </div>
  </section>,
  <div
    className="backdrop"
  />,
]
`;
